lib/helper/WebDriver.js

```
@@ -2569,16 +2569,20 @@ class WebDriver extends Helper {
   async switchTo(locator) {
     this.browser.isInsideFrame = true
     if (Number.isInteger(locator)) {
-      return this.browser.switchToFrame(locator)
+      // @TODO construct array of iFrames and pick "index"
+      let locator1 = new Locator('//iframe[@id="number-frame-1234"]', 'xpath')
```

I guess we forgot to adjust this hardcoded id?

---

I think we need to fetch a list of iframes dynamically and then fetch the corresponding index?

---

well, I'm not having much knowledge on this, but I guess, we shall query the iframe based on the passed locator and switch to that frame if found, otherwise, we shall throw an error as no frame to switch to, what do you think.

---

The locator passed is a number i guess in previous versions that what have returned the frame with that index - we might need to either investigate if this should be possible in the new version or not. Could not find a method to get all iframes on a page - yop might now a way?

---

https://codecept.io/helpers/WebDriver/#switchto looks like it's either a locator or nothing.

---

Then I guess it should be adapted to that together with the tests - I could do that